opm/autodiff/NewtonIterationBlackoilInterleaved.hpp

```diff
@@ -132,13 +132,28 @@ namespace Opm
                         AdditionalObjectDeleter<SeqPreconditioner> >
         constructPrecond(Operator& opA, const Comm& comm) const
         {
+            typedef AdditionalObjectDeleter<SeqPreconditioner> Deleter;
+            typedef std::unique_ptr<ParPreconditioner, Deleter> Pointer;
+            Pointer precond;
+            int ilu_setup_successful = 1;
+            std::string message;
             const double relax = 1.0;
-            SeqPreconditioner* seq_precond= new SeqPreconditioner(opA.getmat(),
+            try{
+                SeqPreconditioner* seq_precond= new SeqPreconditioner(opA.getmat(),
                                                                   relax);
-            typedef AdditionalObjectDeleter<SeqPreconditioner> Deleter;
-            std::unique_ptr<ParPreconditioner, Deleter>
-                precond(new ParPreconditioner(*seq_precond, comm),
-                        Deleter(*seq_precond));
+                precond = Pointer(new ParPreconditioner(*seq_precond, comm),
+                                  Deleter(*seq_precond));
+            }catch(Dune::MatrixBlockError error)
+            {
+                message = error.what();
+                std::cerr<<message<<std::endl;
```

Comment:
If more than one process fail the output can become messed up I think. Could we do something about that? I am fine with leaving it in for now (since I still consider the parallel version experimental), but eventually it should be atomic in some sense. Perhaps now it should also identify the process that is having trouble, for debugging?